crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -257,21 +257,15 @@ where
             Stmt::Global(ast::StmtGlobal { names, range: _ }) => {
                 let ranges: Vec<TextRange> = helpers::find_names(stmt, self.locator).collect();
                 if !self.semantic_model.scope_id.is_global() {
-                    // Add the binding to the current scope.
-                    let context = self.semantic_model.execution_context();
-                    let exceptions = self.semantic_model.exceptions();
-                    let scope = &mut self.semantic_model.scopes[self.semantic_model.scope_id];
                     for (name, range) in names.iter().zip(ranges.iter()) {
                         // Add a binding to the current scope.
-                        let binding_id = self.semantic_model.bindings.push(Binding {
-                            kind: BindingKind::Global,
-                            range: *range,
-                            references: Vec::new(),
-                            source: self.semantic_model.stmt_id,
-                            context,
-                            exceptions,
-                            flags: BindingFlags::empty(),
-                        });
+                        let binding = self.semantic_model.scoped_binding(
+                            *range,
+                            BindingKind::Global,
+                            BindingFlags::empty(),
+                        );
+                        let binding_id = self.semantic_model.bindings.push(binding);
```

Comment:
What's the reason that `scoped_binding` does not perform the `bindings.push` operation? Are there cases where we don't want to push the binding to bindings?

Reply:
Let me revisit this in a separate PR. There's some complicated stuff that happens in `add_binding` whereby we need to determine whether a new binding is shadowing an existing binding, and so we need to create the binding in order to do that check, but can't push it into the scope yet. I'd like to move that entire thing into the semantic model.